backend/algorithms/objects/conditions.py

```diff
@@ -400,10 +417,18 @@ def validate(self, user: User) -> tuple[bool, list[str]]:
 
         validations = [cond.validate(user) for cond in self.conditions]
         # unzips a zipped list - https://www.geeksforgeeks.org/python-unzip-a-list-of-tuples/
-        unlocked, warnings = list(zip(*validations))
-        satisfied = all(unlocked) if self.logic == Logic.AND else any(unlocked)
-
-        return satisfied, sum(warnings, [])  # warnings are flattened
+        wam_warning = []
+        unlocked, all_warnings = list(zip(*validations))
+        for unlocked_cond, warning in validations:
+            if unlocked_cond and len(warning) > 0:
+                wam_warning.append(warning)
```

Comment:
this should be a list comprehension:
```python
wam_warning = sum(warning for unlocked_cond, warning in validations if unlocked_cond, [])
```
this also gets rid of weird issues below with types.

Reply:
(my code may be buggy, take it as a suggestion not as a command lol)